src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -490,11 +521,7 @@ internal override void RequestRun()
 
             if (RunSettings.RunType != RunType.Manual)
             {
-                // TODO for Dynamo 3.0: The boolean "executingTask" that is used here is a make-do fix.
-                // We will be needing a separate variable(boolean flag) to check whether run can be enabled from external applications
-                // and not confuse it with the internal flag RunEnabled which is associated with the Run button in Dynamo. 
-                // Make this RunSettings.RunEnabled private, introduce the new flag and remove the "executingTask" variable. 
-                if ((RunSettings.RunEnabled || executingTask) && !DelayGraphExecution)
```

Comment:
there were some hairy bugs this flag was related to I think either @reddyashish or @RobertGlobant20 worked on this - did you look into the history of why this and the comments exist?

Reply:
Yes. 
Here is the scripture:
1. In the beginning (~2015) the `RunEnabled` flag was introduced as a way to notify UI elements (ex Run button) that the graph was running.  [commit](https://github.com/DynamoDS/Dynamo/commit/41397868fc5a15dfeb272604544fc80cd75a7aad#diff-69f082a38992f2e5879342dcf124febcd476976748fe61a2bc1006041811562cR192)
2. In 2019 the `RunEnabled` flag was extended to also block execution. [commit](https://github.com/DynamoDS/Dynamo/commit/504f53b97deece81aea8f666b03552abd94a4bb8#diff-df9fc71f88d3bf7d4b6700f044ed43c5c6d16e44b80ba2efdef111f75ee16c5fR334)
3. Shortly after that a bug was identified and another flag was introduced `executingTask`. This was again meant to represent that the Graph was currently running or not. [commit](https://github.com/DynamoDS/Dynamo/commit/76c7b618b1668aee9b99571e3e75706c70a14f4a#diff-df9fc71f88d3bf7d4b6700f044ed43c5c6d16e44b80ba2efdef111f75ee16c5fR36)

The requirements (so far) around when a Graph is being run are:
1. **Parts of the UI need to be disabled when a graph is running (ex Run button)**
2. **If a Graph Run has started (i.e GraphUpdate task has been scheduled), other Graph run calls need to be permitted (i.e other GraphUpdate tasks can be created**). An example case would be hooking up a connector to an already connected port. This translates to 2 consecutive Graph runs. 
3. **Host integrators want an API (they are already using` RunEnabled`)  to disable Graphs from being run (not just UI) when the host apps change documents**